src/app/core/api/portal-api.service.ts

```diff
@@ -243,6 +244,31 @@ export class PortalApi extends BaseApi {
     getAssetProfiles(): Observable<IAssetProfile[]> {
         return this.get<IAssetProfile[]>(`${this._apiUrl}/assetprofile/get`);
     }
+
+    getPlantInformations(): Observable<IPlantInformation[]> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation`);
+    }
+
+    getPlantInformation(id: number): Observable<IPlantInformation> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation/${id}`);
+    }
```

Comment:
we don't have GetPlantInformation  by id. This should be by code as the Plant Information table doesn't have id in the first place

Reply:
Kindly see update